### DISCUSSION TITLE
how to use typePolicies to replace updateQuery

### DISCUSSION THREAD

**Discussion by nagaky4** [Akryum/vue-apollo#1287] [Q&A]
Hi everybody,
The **updateQuery** is deprecated, so i try to use **typePolicies**  instead, but the data of fetchMore function are not merge.

When i use **updateQuery**, it work but i got the alert above.
![image](https://user-images.githubusercontent.com/31811450/142767080-eb1dd537-fc3e-447d-9720-b96fc941848b.png)

I update it with **typePolicies**
![image](https://user-images.githubusercontent.com/31811450/142767151-5d92f25b-fe6d-4d60-8274-8018d395bfab.png)
![question](https://user-images.githubusercontent.com/31811450/142767550-af6e7148-1b5e-4048-9186-b25926764c32.png)

Query
![image](https://user-images.githubusercontent.com/31811450/142767370-cc99264c-eef2-425f-89a2-231f1e645b3e.png)

Did i do it wrong?
Could you help me, please.
Thanks.

**Comment by Akryum** [comment]
Try  `return [...existing, ...incoming]`

**Reply by nagaky4** [reply]
hi @Akryum 
I have tried it, but it's not working. 
![image](https://user-images.githubusercontent.com/31811450/142771525-8e8f08fe-45cf-452a-8889-4d9a3564aa7e.png)
![image](https://user-images.githubusercontent.com/31811450/142771399-3ca28866-5f91-4850-ac86-3ad771334921.png)
and the useResult not run
![image](https://user-images.githubusercontent.com/31811450/142771512-4c1899cb-336a-4bd3-8873-0a8eb77108c7.png)

**Reply by Akryum** [reply]
You should write a merge for the field which is the array: `Departements: { fields: { charts: { merge... } } }`

**Reply by nagaky4** [reply]
I have changed it, but nothing happens
![image](https://user-images.githubusercontent.com/31811450/142772655-5bddd6dc-8762-41e1-abf6-13120d69e147.png)
The existing array always empty
![1](https://user-images.githubusercontent.com/31811450/142772746-bdb6268f-e91f-40b3-9bc9-b1f8c62e1764.png)

**Reply by Akryum** [reply]
What is the graphql document?

**Reply by nagaky4** [reply]
Thanks a lot for your support, here is my graphql document.
![image](https://user-images.githubusercontent.com/31811450/142773219-82f684ba-ec8a-45da-ac60-3bc17f56d2bb.png)
![image](https://user-images.githubusercontent.com/31811450/142773322-28b0b1ba-eb9c-4ca9-bded-0899317c507c.png)

**Reply by Akryum** [reply]
Have you configured apollo client to use `'Department' + name` to identify the Departement objects? I don't see an `id` field on it in the query.

**Comment by nagaky4** [comment]
hi @Akryum 
in my graphql document: field `urn` is unique so I add it in `keyFields` and the array can be updated now, **but the pageInfo is not updated**
![image](https://user-images.githubusercontent.com/31811450/142809404-4dfcf53b-7b69-4cec-9588-aa4162ed4db1.png)
![image](https://user-images.githubusercontent.com/31811450/142808761-b124505f-1cb1-4d44-8a99-915da44e9e70.png)
here is the response
![2](https://user-images.githubusercontent.com/31811450/142809223-66fb3aad-8d07-4670-b952-c6f19074fcd6.png)

**Reply by Akryum** [reply]
You probably need a merge policy for the `RestListChart` `pageInfo` field.

**Reply by Akryum** [reply]
(Your pageInfo object doesn't seem to have enough info on its own to be identified by apollo client.)

**Reply by nagaky4** [reply]
I try to do that, it works well, the pageInfo can be updated
![image](https://user-images.githubusercontent.com/31811450/142838619-486ef186-601d-4e29-b6ed-6a254251e329.png)
![3](https://user-images.githubusercontent.com/31811450/142838861-71ca50c9-f58b-469d-910a-8ba90ecb14fe.png)

But when i come back, i got the warning duplicated key, and cannot stop fetchMore, I think the cache was duplicated. 
![image](https://user-images.githubusercontent.com/31811450/142839188-da3f1e1b-8136-4361-b459-5f75762e7dc0.png)

![image](https://user-images.githubusercontent.com/31811450/142839831-08557bc1-d345-447e-9219-06836f28aff6.png)

**Reply by Akryum** [reply]
Why do you use the `relayStylePagination`? You don't use the relay format (which is very strict).

**Reply by nagaky4** [reply]
_> You probably need a merge policy for the `RestListChart` `pageInfo` field._
I try to do that. but the pageInfo ins useResult not updated
![image](https://user-images.githubusercontent.com/31811450/142844783-dd094b8b-62bd-4c54-92cb-60a0d9960373.png)

![4](https://user-images.githubusercontent.com/31811450/142845379-6f5e1302-99d3-4c4f-aeae-47b3168a085e.png)

**Reply by Akryum** [reply]
I see that you have no identifier for the `ResListChart`

**Reply by Akryum** [reply]
Try a `keyFields: ['department', ['urn']]` on it

**Reply by nagaky4** [reply]
@Akryum thanks a lot, it works well. I'm so appreciate 🎉🎉🎉
one more thing if i use **fetchPolicy** : `network-only` 
![image](https://user-images.githubusercontent.com/31811450/142884490-6fe172b0-a1d3-47bb-ba4e-eaad5845ef86.png)
When I come back, the cache will be duplicated, please could you tell me how can I clear the cache first or another way to handle it
![image](https://user-images.githubusercontent.com/31811450/142885079-55c59316-718f-4510-bece-7653aea4390a.png)

**Reply by Akryum** [reply]
You should probably not use `network-only` with `fetchMore`

**Reply by nagaky4** [reply]
only `cache-first` work, isn't it?

**Reply by nagaky4** [reply]
but for the `cache-first`. when i come back. i cannot get new data
